lib/prime.rb

```diff
@@ -132,7 +209,8 @@ def method_added(method) # :nodoc:
   #   Upper bound of prime numbers. The iterator stops after it
   #   yields all prime numbers p <= +ubound+.
   #
-  def each(ubound = nil, generator = EratosthenesGenerator.new, &block)
+  #def each(ubound = nil, generator = EratosthenesGenerator.new, &block)
+  def each(ubound = nil, generator = NextPrimeGenerator.new, &block)
```

Comment:
jzakiya's miller_rabin_test method could use at least one tweak at the end, apply De Morgan's law.

```Ruby
  def miller_rabin_test(b)
    return self >= 2 if self <= 3 
    # I would avoid creating a temp Array to check only 2 cases
    return false unless [1,5].include?(self % 6)
    n = d = self - 1 # self - 1 is pred, perhaps less readable
    d >>= 1 while d.even?
    y = b.to_bn.mod_exp(d, self)
    while d != n && y != n && y != 1
      y = y.to_bn.mod_exp(2, self)
      d <<= 1
    end
    #return false if y != n && d.even?
    #true
    y == n || d.odd?
  end
```